libs/chonky/src/embedding/multi_modal_embedding.rs

```diff
@@ -0,0 +1,338 @@
+use base64::{Engine as _, engine::general_purpose};
+use error_stack::{Report, ResultExt as _};
+use image::DynamicImage;
+use reqwest::{
+    Client,
+    header::{AUTHORIZATION, CONTENT_TYPE, HeaderMap, HeaderValue},
+};
+use serde_json::{Value, json};
+
+use crate::{
+    ChonkyError, DocumentEmbeddings, Embedding, StructuralEmbedding, StructuralMetadata,
+    pdf_segmentation::ExtractedTable,
+};
+
+fn get_vertex_access_token() -> Result<String, Report<ChonkyError>> {
+    Ok(String::from_utf8(
+        std::process::Command::new("gcloud")
+            .args(["auth", "print-access-token"])
+            .output()
+            .change_context(ChonkyError::VertexAPI)?
+            .stdout,
+    )
+    .change_context(ChonkyError::VertexAPI)?
+    .trim()
+    .to_owned())
+}
+
+fn base64_json(image_data: Vec<u8>) -> Result<String, Report<ChonkyError>> {
+    let base64_encoded_img = general_purpose::STANDARD.encode(image_data);
+    let formatted_json = serde_json::to_string_pretty(&json!({
+        "instances": [
+            {
+                "image": {
+                    "bytesBase64Encoded": base64_encoded_img
+                }
+            }
+        ]
+    }))
+    .change_context(ChonkyError::VertexAPI)?;
+
+    // add newline that is appended by yarn in the snapshot
+    Ok(format!("{formatted_json}\n"))
+}
+
+fn text_json(text: &[String]) -> Result<String, Report<ChonkyError>> {
+    //merge all text into one without seperator for now
+    let mut text = text.join("");
+    text.truncate(1000);
```

Comment:
This is mentioned in a comment when calling this function, but basically the Vertex API only allows character counts of I believe 1028 characters, thus since chunking is not being done yet, we just truncate the text to follow this.